test/oraclize/OracleHubOraclize.js

```diff
@@ -136,4 +155,159 @@ contract('OracleHubOraclize', function(accounts) {
+    await oracleHub.requestQuery(deployedMarketContract.address, oracleDataSource, oracleQuery, 0, {
+      from: accounts[1]
+    });
+
+    let oraclizeCallbackGasCost = await oracleHub.QUERY_CALLBACK_GAS.call();
+    let contractSettledEvent = deployedMarketContract.ContractSettled();
+
+    contractSettledEvent.watch(async (err, response) => {
+      if (err) {
+        console.log(err);
+      }
+      assert.equal(response.event, 'ContractSettled');
+      txReceipt = await web3.eth.getTransactionReceipt(response.transactionHash);
+      // console.log('Oraclize callback gas used : ' + txReceipt.gasUsed);
+      assert.isBelow(
+        txReceipt.gasUsed,
+        oraclizeCallbackGasCost.toNumber(),
+        'Callback tx claims to have used more gas than allowed!'
+      );
+      contractSettledEvent.stopWatching();
+    });
+
+    const waitForContractSettledEvent = ms =>
+      new Promise((resolve, reject) => {
+        const check = () => {
+          if (txReceipt) resolve();
+          else if ((ms -= 1000) < 0) reject(new Error('Oraclize time out!'));
+          else {
+            setTimeout(check, 1000);
+          }
+        };
+        setTimeout(check, 1000);
+      });
+
+    await waitForContractSettledEvent(60000);
+    assert.notEqual(
+      txReceipt,
+      null,
+      'Oraclize callback did not arrive. Please increase QUERY_CALLBACK_GAS!'
+    );
+  });
+
+  describe('requestOnDemandQuery()', function() {
+    it('should fail if not eth is sent', async function() {
+      let error = null;
+      try {
+        await oracleHub.requestOnDemandQuery(marketContract.address, { from: accounts[0] });
+      } catch (err) {
+        error = err;
+      }
+
+      assert.instanceOf(error, Error, 'query did not fail');
+    });
+
+    it('should emit OraclizeQueryRequested', async function() {
+      let txReceipt = null;
+
+      await oracleHub.requestOnDemandQuery(marketContract.address, {
+        from: accounts[0],
+        value: web3.toWei(1)
+      });
+
+      let oraclizeQueryRequestEvent = oracleHub.OraclizeQueryRequested();
+      oraclizeQueryRequestEvent.watch(async (err, eventLogs) => {
+        if (err) {
+          console.log(err);
+        }
+        assert.equal(eventLogs.event, 'OraclizeQueryRequested');
+        txReceipt = await web3.eth.getTransactionReceipt(eventLogs.transactionHash);
+        oraclizeQueryRequestEvent.stopWatching();
+      });
+
+      const waitForQueryEvent = ms =>
+        new Promise((resolve, reject) => {
+          const check = () => {
+            if (txReceipt) resolve();
+            else if ((ms -= 1000) < 0) reject(new Error('Oraclize time out!'));
+            else {
+              setTimeout(check, 1000);
+            }
+          };
+          setTimeout(check, 1000);
+        });
+
+      await waitForQueryEvent(60000);
+      assert.notEqual(
+        txReceipt,
+        null,
+        'Oraclize callback did not inovke marketContract.oracleCallback()'
```

Comment:
maybe I missed where we are ensuring the oracleCallBack occured.

Reply:
Oops! I'll update the error message